zerver/views/realm.py

```diff
@@ -93,4 +95,7 @@ def update_realm(request, user_profile, name=REQ(validator=check_string, default
     if waiting_period_threshold is not None and realm.waiting_period_threshold != waiting_period_threshold:
         do_set_realm_waiting_period_threshold(realm, waiting_period_threshold)
         data['waiting_period_threshold'] = waiting_period_threshold
+    if notifications_stream_name is not None and realm.notifications_stream.name != notifications_stream_name:
+        do_set_realm_notifications_stream(realm, notifications_stream_name)
+        data['notifications_stream_name'] = notifications_stream_name
```

Comment:
you should add test logic to `test_realm.py` for this feature, similar to what's there, to give this code test coverage.

Also use `access_stream` to fetch the stream here.